### DISCUSSION TITLE
Progressive Data Display end Error form a Single Provider in Flutter

### DISCUSSION THREAD

**Discussion by MichaelLucca** [rrousselGit/riverpod#3372] [Q&A] [answered]
Description
Currently, I'm using a single provider in Flutter to fetch data from multiple API calls within my application. However, I'm encountering difficulty in managing the progressive display of Data and Errors in my UI. 

I want to display portions of the UI as data is received, rather than having to wait for all API calls to complete.
I want to display portions of the UI as data is received, and if the APi call fails I want to show an Error UI

The Example Provider and Widgets are the same from the last question:
![307059002-27254432-3bf0-4c69-9bf9-7a36937570b3](https://github.com/rrousselGit/riverpod/assets/89570547/c0aa2c4f-2d23-4eae-aeff-e1cd07137395)

![307059276-4a813614-c4b0-4d94-8222-d3e248099dc4](https://github.com/rrousselGit/riverpod/assets/89570547/0e47bd07-6e88-4a93-9328-dfa76bfec464)

**Comment by snapsl** [comment]
Your images don't show how you handle the `AsyncError` case. Furthermore, please implement the suggestion from #3357.

**Reply by MichaelLucca** [reply]
@snapsl sorry, I've putted the wrong screenshots. This is the actual situation:
![Screenshot 2024-02-28 alle 10 36 28](https://github.com/rrousselGit/riverpod/assets/89570547/fb2eec81-1540-4157-af49-0750c8a7d765)
![Screenshot 2024-02-28 alle 10 37 08](https://github.com/rrousselGit/riverpod/assets/89570547/6d02d4e5-9a5a-4986-a8a2-2e514594511b)

**Reply by snapsl** [reply]
Can you reformulate what the issue is?

**Reply by MichaelLucca** [reply]
Basically, I have a provider that makes multiple API calls. I want to be able to display the data received from the first API call and simultaneously the error because the second API call doesn't work. If I emit the "error" state, the entire UI would show an error, whereas I want to be able to display data and errors separately.

**Reply by rrousselGit** [reply, marked as answer]
You should instead have multiple providers for this.

**Reply by snapsl** [reply]
So this is just a suggestion...

Instead of trying to combine both providers and handle for the different AsyncStates in the newly created `StoreItemDetails` provider, you could handle both providers separately in the UI.

So wrap `storeitemdetailsprovider` inside `storeitemprovider` and fill the data/load/errors cases accordingly.

**Reply by MichaelLucca** [reply]
Thanks @snapsl @rrousselGit you're both right, I can achieve the behaviour by doing so instead of having a provider acting like a controller 🙏🏻
